workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Systems/SpatialOSReceiveSystemTests.cs

```diff
@@ -0,0 +1,738 @@
+
+                var count = 0;
+                Assert.DoesNotThrow(() => { count = responses.Responses.Count; });
+                Assert.AreEqual(1, count);
+
+                var response = responses.Responses[0];
+
+                Assert.AreEqual(emptyRequest, response.RequestPayload);
+                Assert.AreEqual(context, response.Context);
+                Assert.AreEqual(TestCommandRequestId, response.RequestId);
+                Assert.AreEqual(wrappedOp.Op, response.Op);
+            }
+        }
+
+        [Test]
+        public void OnDeleteEntityRespoonse_should_error_if_request_id_not_found()
+        {
+            using (var wrappedOp = WorkerOpFactory.CreateDeleteEntityResponseOp(TestCommandRequestId))
+            {
+                logDispatcher.Expect(new ExpectedLog(LogType.Error, LoggingUtils.LoggerName, "RequestId",
+                    "Command Type"));
+                receiveSystem.OnDeleteEntityResponse(wrappedOp.Op);
+                logDispatcher.AssertAgainstExpectedLogs();
+            }
+        }
+
+        [Test]
+        public void OnDeleteEntityResponse_should_log_if_corresponding_entity_not_found()
+        {
+            var emptyRequest = new WorldCommands.DeleteEntity.Request();
+
+            deleteEntityStorage.CommandRequestsInFlight.Add(TestCommandRequestId,
+                new CommandRequestStore<WorldCommands.DeleteEntity.Request>(Entity.Null,
+                    emptyRequest, null, TestCommandRequestId));
+
+            using (var wrappedOp = WorkerOpFactory.CreateDeleteEntityResponseOp(TestCommandRequestId))
+            {
+                logDispatcher.Expect(new ExpectedLog(LogType.Log, LoggingUtils.LoggerName, "Op"));
+                receiveSystem.OnDeleteEntityResponse(wrappedOp.Op);
+                logDispatcher.AssertAgainstExpectedLogs();
+            }
+        }
+
+        [Test]
+        public void OnEntityQueryResponse_should_add_received_responses_to_entity()
+        {
+            var entity = SetupTestEntity();
+
+            var emptyRequest = new WorldCommands.EntityQuery.Request();
+            var context = "Some context";
+
+            entityQueryStorage.CommandRequestsInFlight.Add(TestCommandRequestId,
+                new CommandRequestStore<WorldCommands.EntityQuery.Request>(entity,
+                    emptyRequest, context, TestCommandRequestId));
+
+            using (var wrappedOp = WorkerOpFactory.CreateEntityQueryResponseOp(TestCommandRequestId))
+            {
+                receiveSystem.OnEntityQueryResponse(wrappedOp.Op);
+
+                Assert.IsTrue(entityManager.HasComponent<WorldCommands.EntityQuery.CommandResponses>(entity));
+
+                var responses = entityManager.GetComponentData<WorldCommands.EntityQuery.CommandResponses>(entity);
+
+                var count = 0;
+                Assert.DoesNotThrow(() => { count = responses.Responses.Count; });
```

Comment:
Why not check against null instead of `DoesNotThrow`?

Reply:
I can't - the `Provider`s don't return null - they throw an `ArgumentException` that the handle was not found

Reply:
Ah, yes. 👍 resolving